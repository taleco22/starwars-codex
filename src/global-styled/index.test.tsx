import React from "react";
import renderer from "react-test-renderer";
import "../setupTest";
import { globalButton, globalWindow } from "./index";

const Button = globalButton;
const Window = globalWindow;

describe("globalWindow", () => {
  it("have animation", () => {
    expect(renderer.create(<Window shake />).toJSON()).toMatchSnapshot();
  });
  it("have NO animation", () => {
    expect(renderer.create(<Window />).toJSON()).toMatchSnapshot();
  });
});

describe("globalButton", () => {
  it("render default style rules", () => {
    expect(renderer.create(<Button />).toJSON()).toMatchSnapshot();
  });

  it("enable pagination styling", () => {
    expect(renderer.create(<Button paginate />).toJSON()).toMatchSnapshot();
  });

  it("enable off button styling", () => {
    expect(renderer.create(<Button off />).toJSON()).toMatchSnapshot();
  });
});
