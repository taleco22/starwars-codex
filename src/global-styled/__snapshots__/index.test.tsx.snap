// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`globalButton enable off button styling 1`] = `
.c0 {
  -webkit-text-decoration: none;
  text-decoration: none;
  text-transform: uppercase;
  color: #fff;
  font-weight: bold;
  background-color: #323535;
  border: none;
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 1rem 0;
  opacity: 0.7;
  font-size: 1.6rem;
  background-color: #111;
}

.c0:hover,
.c0:focus {
  opacity: 1;
  background-image: linear-gradient( 90deg, #660a2d 0%, #870a2d 5%, #53121a 50%, #870a2d 95%, #660a2d 100% );
}

.c0 h1 {
  font-size: 1.6rem;
}

@media (min-width:768px) {
  .c0 {
    max-width: 40vw;
  }
}

<button
  className="c0"
/>
`;

exports[`globalButton enable pagination styling 1`] = `
.c0 {
  -webkit-text-decoration: none;
  text-decoration: none;
  text-transform: uppercase;
  color: #fff;
  font-weight: bold;
  background-color: #323535;
  border: none;
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 1rem 0;
  opacity: 0.7;
  font-size: 1.6rem;
  color: #ffe81f;
  background-color: unset;
  background-color: unset;
  padding: 0.5rem;
  margin: 0 0.5rem;
}

.c0:hover,
.c0:focus {
  opacity: 1;
  background-image: linear-gradient( 90deg, #660a2d 0%, #870a2d 5%, #53121a 50%, #870a2d 95%, #660a2d 100% );
}

.c0 h1 {
  font-size: 1.6rem;
}

@media (min-width:768px) {
  .c0 {
    max-width: 40vw;
  }
}

<button
  className="c0"
/>
`;

exports[`globalButton render default style rules 1`] = `
.c0 {
  -webkit-text-decoration: none;
  text-decoration: none;
  text-transform: uppercase;
  color: #fff;
  font-weight: bold;
  background-color: #323535;
  border: none;
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 1rem 0;
  opacity: 0.7;
  font-size: 1.6rem;
}

.c0:hover,
.c0:focus {
  opacity: 1;
  background-image: linear-gradient( 90deg, #660a2d 0%, #870a2d 5%, #53121a 50%, #870a2d 95%, #660a2d 100% );
}

.c0 h1 {
  font-size: 1.6rem;
}

@media (min-width:768px) {
  .c0 {
    max-width: 40vw;
  }
}

<button
  className="c0"
/>
`;

exports[`globalWindow have NO animation 1`] = `
.c0 {
  position: relative;
  width: 100vw;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  height: 100vh;
}

<main
  className="c0"
/>
`;

exports[`globalWindow have animation 1`] = `
.c0 {
  position: relative;
  width: 100vw;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  height: 100vh;
  -webkit-animation: ibKLYw 1.25s;
  animation: ibKLYw 1.25s;
}

<main
  className="c0"
/>
`;
